doc/Client-Portal Protocol.md

```diff
@@ -350,38 +356,44 @@ This a format for client reconnects.
      message: string(Message)
     }
 
-### 3.3.7 Participant Starts Publishing a WebRTC Stream to Room
+### 3.3.7 Participant Starts Publishing a Stream to Room
 **RequestName**: “publish”<br>
 
 **RequestData**: The PublicationRequest object with following definition:
 
+```
   object(PublicationRequest)::
     {
-     media: object(WebRTCMediaOptions),
-     attributes: object(ClientDefinedAttributes)
+     media: object(WebRTCMediaOptions) | null,
+     data: true | false,
+     transport: object(TransportOptions),
```

Comment:
Previous publications can be "webrtc", "streaming" or "sip".
Are the "data" and "transport" fields for "webrtc" or "quic-p2p" only?
I'm not sure, is a quic-connection sending media data a "quic-p2p" publication or "webrtc" publication?

Reply:
The idea of adding a `transport` is to make it possible to reuse transport channels. I'm not sure if SIP or other streaming transport channels could be reused.

Sending media data through QUIC connection is a "quic-p2p" publication, although it's still belongs to WebRTC right now. As we decided to create a new agent for QUIC instead of put it in WebRTC agent, we need a new type of QUIC connections.

Reply:
I mean the description here is not clear.
Can "webrtc" transport publications have "data" field? The "data" is only needed by newly added "quic-p2p" publications?
Can these fields be optional? In my understanding, the "data" and "media" fields are exclusive, one publication can not have them both. Is a publication allowed with "media" null and "data" false? And a "webrtc" transport publication's "media" can not be null? I think you'd better add more comment in document or adjust the format to make it more clear.

Reply:
From protocol perspective, a WebRTC publication could have `data:true` if data channel is supported, but it's not implemented right now. I'll add more description to it.